Kuadrant/limitador-operator
# Updating status

This PR enhance the `Limitador.Status` with a new format, separating the _Host_ from the _Ports_.  It also checks if the instance of limitador is running and manages the readiness with _apimachinery API Conditions_

*Verification Steps:*
1. Lunch the cluster setup with 
```bash
make local-setup
```
2. When ready, create a new namespace:
```bash
kubectl create namespace "limitador-test"
```
3. Then, apply the following Limitador CR:
```bash
k -n limitador-test apply -f - <<EOF
apiVersion: limitador.kuadrant.io/v1alpha1
kind: Limitador
metadata:
  name: limitador-sample
spec:  {}
EOF
```

4. Then after the operator created the deployment, service,configmaps, etc... you could check the status of the limitador CR:
```bash
k -n limitador-test get limitador/limitador-sample
```
```yaml
...
Status:
  Conditions:
    Last Transition Time:  2022-07-21T16:46:23Z
    Message:             ""
    Reason:                LimitadorInstanceRunning
    Status:                 True
    Type:                   Ready
  Service:
    Host:  limitador-sample.limitador-test.svc.cluster.local
    Ports:
      Grpc:  8081
      Http:  8080
```
